projects/cm_mcu/I2CCommunication.c

```diff
@@ -114,14 +113,14 @@ int apollo_i2c_ctl_reg_r(uint8_t device, uint8_t address, uint8_t nbytes_addr, u
   tSMBusStatus* p_status = eStatus[device];
 
   configASSERT(smbus != NULL);
-  uint8_t reg_address[MAX_BYTES_ADDR];
-    for (int i = 0; i < MAX_BYTES_ADDR; ++i) {
-        reg_address[i] = (packed_reg_address >> (i) * 8) & 0xFFUL;
-      }
+  uint8_t reg_address[nbytes_addr];
```

Comment:
sorry I think i miscommunicated my requested change. With this change you're now making a c VLA (variable length array). I.e., an array with dimensions not defined at compile time. while this is technically supported in C (and your compiler will not complain) it is generally though of as a bad part of the standard (see e.g. [this page discussing its pitfalls](https://blog.joren.ga/vla-pitfalls). Can you just define a fixed length array but only send the appropriate number of bytes.

Reply:
Thanks for the info about c VLA. I was not sure if "define a fixed length array but only send the appropriate number of bytes" will have the same syntax as what I did previously in which data[1] won't be written for a one-byte register address.

Reply:
my thought was that for the data array it would look like this
```
data[0] = addr[0]
data[1] = addr[1]
data[2] = first byte of packed data
....
```
for 2 byte addresses and
```
data[0] = addr[0]
data[1] = first byte of packed data
....
```
for one-byte addresses

Reply:
If we want to avoid VLA, I understand that data array should be declared as ` uint8_t data[MAX_BYTE_ADDR+MAX_BYTES];` From your example, do you mean that data[MAX_BYTE_ADDR+MAX_BYTES -1] might be empty for one-byte address.